src/app/account/mfa/sms/page.tsx

```diff
@@ -0,0 +1,199 @@
+'use client'
+
+import * as React from 'react'
+import { useUser, useClerk } from '@clerk/nextjs'
+
+import { Button, Flex, Title } from '@mantine/core'
+import { BackupCodeResource, PhoneNumberResource } from '@clerk/types'
+import Link from 'next/link'
+
+// Display phone numbers reserved for MFA
+const ManageMfaPhoneNumbers = () => {
+    const { user } = useUser()
+
+    if (!user) return null
+
+    // Check if any phone numbers are reserved for MFA
+    const mfaPhones = user.phoneNumbers
+        .filter((ph) => ph.verification.status === 'verified')
+        .filter((ph) => ph.reservedForSecondFactor)
+        .sort((ph: PhoneNumberResource) => (ph.defaultSecondFactor ? -1 : 1))
+
+    if (user.phoneNumbers.length === 0) {
+        return <p>There are currently no phone numbers on your account.</p>
+    }
+
+    return (
+        <>
+            <h2>Phone numbers reserved for MFA</h2>
+            <ul>
+                {mfaPhones.map((phone) => {
+                    return (
+                        <li key={phone.id} style={{ display: 'flex', gap: '10px' }}>
+                            <p>
+                                {phone.phoneNumber} {phone.defaultSecondFactor && '(Default)'}
+                            </p>
+                            <div>
+                                <Button onClick={() => phone.setReservedForSecondFactor({ reserved: false })}>
+                                    Disable for MFA
+                                </Button>
+                            </div>
+
+                            {!phone.defaultSecondFactor && (
+                                <div>
+                                    <Button onClick={() => phone.makeDefaultSecondFactor()}>Make default</Button>
+                                </div>
+                            )}
+
+                            {user.phoneNumbers.length > 1 && (
+                                <div>
+                                    <Button onClick={() => phone.destroy()}>Remove from account</Button>
+                                </div>
+                            )}
+                        </li>
+                    )
+                })}
+            </ul>
+            You have enabled MFA on your account
+            <Link href="/">
+                <Button>Return to homepage</Button>
+            </Link>
+        </>
+    )
+}
+
+// Display phone numbers that are not reserved for MFA
+const ManageAvailablePhoneNumbers = () => {
+    const { user } = useUser()
+
+    if (!user) return null
+
+    // Check if any phone numbers aren't reserved for MFA
+    const availableForMfaPhones = user.phoneNumbers
+        .filter((ph) => ph.verification.status === 'verified')
```

Comment:
general nitpick of preferring to be verbose with the callback arguments ie `ph` -> `phoneNumber` or even `phone` just for legibility 🤔 but this is entirely subjective

Reply:
yah that's kinda weird anyway, why are there two filters?  this code is mostly from the clerk docs btw. I rewrote and moved it inline to the component